### PR TITLE
Fix error stack for some fetch calls.

### PR DESCRIPTION
**Pull request by slontis**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: crypto/evp/evp_pbe.c [hunk 1]
```diff
@@ -117,10 +117,12 @@ int EVP_PBE_CipherInit_ex(ASN1_OBJECT *pbe_obj, const char *pass, int passlen,
         passlen = strlen(pass);
 
     if (cipher_nid != -1) {
+        (void)ERR_set_mark();
         cipher = cipher_fetch = EVP_CIPHER_fetch(libctx, OBJ_nid2sn(cipher_nid), propq);
         /* Fallback to legacy method */
         if (cipher == NULL)
             cipher = EVP_get_cipherbynid(cipher_nid);
+        (void)ERR_pop_to_mark();
```
Comment (levitte) [inline crypto/evp/evp_pbe.c:125]: It doesn't seem right to do this unconditionally here

File: crypto/evp/p5_crpt2.c [hunk 2]
```diff
@@ -141,15 +141,18 @@ int PKCS5_v2_PBE_keyivgen_ex(EVP_CIPHER_CTX *ctx, const char *pass, int passlen,
         goto err;
     }
 
+    (void)ERR_set_mark();
     cipher = cipher_fetch = EVP_CIPHER_fetch(libctx, ciph_name, propq);
     /* Fallback to legacy method */
     if (cipher == NULL)
-        cipher = EVP_get_cipherbyname(ciph_name);
+        cipher = EVP_get_cipherbyname(ciph_name);;
```
Comment (levitte) [inline crypto/evp/p5_crpt2.c:148]: Why the extra semi?

Comment (slontis) [reply crypto/evp/p5_crpt2.c]: typo..

**Review by paulidale — APPROVED** [review]

**Review by levitte — APPROVED** [review]

**Comment by slontis** [comment]
Merged to master.

**Pull request closed without merging**